### ISSUE TITLE
NAVMESH doesn't generate in modular sections

### ISSUE THREAD

**Issue by mav3r1ck1981** [TheGameCreators/GameGuruRepo#5723]
so, the overwhelming majority of my levels are interior and these are built using modular packs with each individual section imported via the importer. Having spent some time building a new level for a competition entry, i could not get any characters (both stock. custom or those using the NPC Controller) would interact at all with the player. They would run through their idle animation, but they don't walk, attack, run away or find cover. Basically any of the things you require an NPC to do. 

Playing the level in test game with "Show Navigation Debug Visuals" ticked on, it was immediately obvious that the NAVMESH wasn't being generated within the modules. However the Navmesh WAS being generated on the terrain below the modules in a flat yellow square. 

![image](https://github.com/user-attachments/assets/59f66727-2c84-462e-bb9c-ab3f3e5956e8)

I figured with Necryms help that this could be because the entire level was suspended in the air and locked, it had no contact with the terrain/ground. So i added 2 ramps, beginning and end of the level, to provide that link and tried again. Still nothing.

Next, i took an individual corridor section, and laid it on the ground in a test area, and unfortunately, the same thing happens.

![image](https://github.com/user-attachments/assets/7592da1c-8249-43c2-8753-102b64790328)

![image](https://github.com/user-attachments/assets/9b97e72f-353e-4462-8201-004183584391)

The only difference in these modules and perhaps others ive used in other levels is these are higher poly, this corridor for example is 200K poly

**Comment by fearlesswee** [comment]
My thought may be that the corridor is either too narrow for the navmesh to see it as traversable, and/or the complex shape/collision of the floor is causing the navmesh to similarly not see a valid traversable area there. Try scaling up the piece massively, just as a test, to rule out it being too narrow.

The modular sections from say, the DLC packs, work just fine, but they tend to have flat, level floors, and have more space inside.

**Comment by mav3r1ck1981** [comment]
As a work around, ive added multiple invisible planes to the floors with collision off and have successfully got a navmesh of sorts to generate. see below:

![image](https://github.com/user-attachments/assets/786eb086-ca97-42d1-9e8f-cfe6b9e9cce6)

unfortunately there even though there is a plane set up as above, navmesh still does not generate in some areas.

![image](https://github.com/user-attachments/assets/63a78a39-28e7-433b-8ab2-3f912b9ace0e)

Adding floor decals has made a small area generate, but this is not an ideal solution.

Im going to make an assumption that the rocky corridors are either too high poly, or the meshes are too clustered for navmesh to be generated, but the simple flat plane negates that.

Id like to make a suggestion that as a (near) future addition we could have a NAVMESH brush to physically paint where we need the navmesh to be generated. As GGMax becomes increasingly stable and complex, allowing for more detailed maps, this feels like an important addition.

**Comment by LeeBamberTGC** [comment]
@mav3r1ck1981 Yes I would 'guess' that the model you imported is giving the automated nav mesh generator a hard time, and high polygon count would do that.  Can you email me that massive interior model in its raw and imported states and I will see what the nav mesh is making of it. In the meantime I have retitled and classified this as a feature request as I agree there will be a need for finer grain and manual application of navmeshes without hacking in flat floors that would spoil your game visuals.
